dataplane/linux/wireguard_mgr.go

```diff
@@ -60,33 +66,57 @@ func (m *wireguardManager) OnUpdate(protoBufMsg interface{}) {
 		m.wireguardRouteTable.EndpointRemove(msg.Hostname)
 	case *proto.RouteUpdate:
 		log.WithField("msg", msg).Debug("RouteUpdate update")
-		if msg.Type != proto.RouteType_REMOTE_WORKLOAD {
+		switch msg.Type {
+		case proto.RouteType_LOCAL_WORKLOAD:
+			if cidr := ip.MustParseCIDROrIP(msg.Dst); cidr == nil {
+				log.Errorf("error parsing RouteUpdate CIDR: %s", msg.Dst)
+			} else if cidr.Version() == 4 {
+				log.Debugf("Update for IPv4 CIDR: %s", cidr)
+				m.localCIDRs.Add(cidr)
+				m.wireguardRouteTable.LocalWorkloadCIDRAdd(cidr)
+			}
+		case proto.RouteType_REMOTE_WORKLOAD:
```

Comment:
A CIDR can change from a RouteType_LOCAL_WORKLOAD to something else and in that case you'll get an update rather than a delete followed by an add.  That means you also need to do deletion processing for route updates where the type is no RouteType_LOCAL_WORKLOAD.

Reply:
Ooh, that also applies to RouteType_REMOTE_WORKLOAD; you might get a CIDR that changes from being RouteType_REMOTE_WORKLOAD to something else too.